src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,178 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService;
+    private StorageDao storageDao;
+
+    @BeforeAll
+    public void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void registerWith_correctUserData_ok() {
+        User user1 = new User();
+        user1.setAge(28);
+        user1.setLogin("Vasiliy");
+        user1.setPassword("0123459876");
+        User user2 = new User();
+        user2.setAge(56);
+        user2.setLogin("Svitlana");
+        user2.setPassword("qwerty123");
+        User user3 = new User();
+        user3.setAge(33);
+        user3.setLogin("Penelopa");
+        user3.setPassword("kruzlolqwer");
+        User actualUser1 = registrationService.register(user1);
+        User actualUser2 = registrationService.register(user2);
+        User actualUser3 = registrationService.register(user3);
+        assertEquals(user1, actualUser1);
+        assertEquals(user2, actualUser2);
+        assertEquals(user3, actualUser3);
+    }
+
+    @Test
+    void registerWith_limitUserData_ok() {
```

Comment:
cannot understand difference between this test and void registerWith_correctUserData_ok()

Reply:
It test edge cases as mentioned in the checklist.md